src/s1reader/s1_annotation.py

```diff
@@ -7,20 +7,78 @@
 
 import datetime
 import os
-import lxml.etree as ET
 import zipfile
 
+from types import SimpleNamespace
+
+import lxml.etree as ET
 import numpy as np
 
-from scipy.interpolate import InterpolatedUnivariateSpline, interp1d
+from isce3.core import speed_of_light
 from packaging import version
+from scipy.interpolate import InterpolatedUnivariateSpline, interp1d
 
-from isce3.core import speed_of_light
 
 # Minimum IPF version from which the S1 product's Noise Annotation
 # Data Set (NADS) includes azimuth noise vector annotation
 min_ipf_version_az_noise_vector = version.parse('2.90')
 
+# Minimum IPF version from which the RFI information gets available
+# source: "Sentinel-1: Using the RFI annotations", reference no: MPC-0540,
+# URL: (https://sentinel.esa.int/documents/247904/1653442/
+#       DI-MPC-OTH-0540-1-0-RFI-Tech-Note.pdf)
+RFI_INFO_AVAILABLE_FROM = version.Version('3.40')
+
+# Dictionary of the fields in RFI information, and their data type castor
+dict_datatype_rfi={
+    "swath": str,
+    "azimuthTime": lambda T: datetime.datetime.strptime(T, '%Y-%m-%dT%H:%M:%S.%f'),
+    "inBandOutBandPowerRatio": float,
+    "percentageAffectedLines": float,
+    "avgPercentageAffectedSamples": float,
+    "maxPercentageAffectedSamples": float,
+    "numSubBlocks": int,
+    "subBlockSize": int,
+    "maxPercentageAffectedBW": float,
+    "percentageBlocksPersistentRfi": float,
+    "maxPercentageBWAffectedPersistentRfi": float
+}
+
+
+def element_to_dict(elem_in: ET, dict_tree: dict = None):
+    '''
+    Recursively parse the element tree,
+    return the results as SimpleNameSpace
+
+    Parameters
+    ----------
+    elem_in: ElementTree
+        Input element tree object
+    dict_tree: dict
+        Dictionary to be populated
+
+    Returns
+    -------
+    dict_tree: dict
+        A populated dictionary by `elem_in`
+    '''
+    if dict_tree is None:
+        dict_tree = {}
+    key_elem = elem_in.tag
+    child_elem = list(elem_in.iterchildren())
+
+    if len(child_elem) == 0:
+        # Reached the tree end
+        text_elem = elem_in.text
+        dict_tree[key_elem] = dict_datatype_rfi[key_elem](text_elem)
```

Comment:
If they update their metadata with a new field, I think this line will break. You probably want to test if `key_elem` is a member of `dict_datatype_rfi.keys()`. If not, you probably want to print a warning message so we know we may need to add a new element to `dict_datatype_rfi`.

Reply:
Good idea. I've added the warning message in the case above. Also, the values will be treated as `str` when there is no castor defined in `dict_datatype_rfi`